engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```
@@ -181,11 +181,63 @@ Type getType() {
     return eigentype;
   }
 
+  @ExportMessage
+  boolean hasMetaObject() {
+    return true;
+  }
+
+  @ExportMessage
+  Type getMetaObject() {
+    return getType();
+  }
+
+  @ExportMessage
+  Object getMetaParents() {
+    assert supertype != null;
```

Seems like this assertion can be violated if someone calls `getMetaParents` when `hasMetaParents` returned false.

In that case we will throw an assertion error or usually when running without `-ea` we'll return some array with a `null`.

I think according to the `getMetaParents` contract defined in `InteropLibrary` we should instead throw `UnsupportedMessageException`.